In [9]:
import pandas as pd
import numpy as np

<h1>Загрузка данных</h1>

In [35]:
data = pd.read_csv('Top_650_FIFA.csv', index_col=0) # index_col=0 - чтобы не было доп колонки Unnamed
data

short_name                                 long_name  overall  \
0             L. Messi            Lionel Andrés Messi Cuccittini       93   
1       R. Lewandowski                        Robert Lewandowski       92   
2    Cristiano Ronaldo       Cristiano Ronaldo dos Santos Aveiro       91   
3            Neymar Jr             Neymar da Silva Santos Júnior       91   
4         K. De Bruyne                           Kevin De Bruyne       91   
..                 ...                                       ...      ...   
647         T. Malacia                            Tyrell Malacia       77   
648      Wilson Manafá               Wilson Miguéis Manafá Jancó       77   
649          J. Clauss                           Jonathan Clauss       77   
650             Galeno  Wenderson Rodrigues do Nascimento Galeno       77   
651       J. Lascelles                          Jamaal Lascelles       77   

     age         dob  height_cm  weight_kg club_position  Position  \
0     34  1987-06-24        170         72            RW   Forward   
1     32  1988-08-21        185         81            ST   Forward   
2     36  1985-02-05        187         83            ST   Forward   
3     29  1992-02-05        175         68            LW   Forward   
4     30  1991-06-28        181         70           RCM  Midfield   
..   ...         ...        ...        ...           ...       ...   
647   21  1999-08-17        169         67            LB  Defender   
648   26  1994-07-23        177         69            LB  Defender   
649   28  1992-09-25        178         67           RWB  Defender   
650   23  1997-10-22        179         62            LM  Midfield   
651   27  1993-11-11        188         83            CB  Defender   

               club_name  ... shooting passing  dribbling defending  physic  \
0    Paris Saint-Germain  ...     92.0    91.0       95.0      34.0    65.0   
1      FC Bayern München  ...     92.0    79.0       86.0      44.0    82.0   
2      Manchester United  ...     94.0    80.0       88.0      34.0    75.0   
3    Paris Saint-Germain  ...     83.0    86.0       94.0      37.0    63.0   
4        Manchester City  ...     86.0    93.0       88.0      64.0    78.0   
..                   ...  ...      ...     ...        ...       ...     ...   
647            Feyenoord  ...     56.0    64.0       75.0      72.0    73.0   
648             FC Porto  ...     60.0    67.0       76.0      70.0    69.0   
649  Racing Club de Lens  ...     66.0    70.0       76.0      67.0    71.0   
650             SC Braga  ...     70.0    66.0       82.0      32.0    66.0   
651     Newcastle United  ...     35.0    51.0       56.0      77.0    79.0   

     attacking skill  movement  power  mentality  
0         85.8  94.0      90.2   77.8  73.833333  
1         86.0  81.4      81.6   84.8  80.666667  
2         87.6  83.6      85.4   87.2  74.333333  
3         80.6  89.2      90.2   71.8  77.000000  
4         81.4  88.0      80.0   81.6  82.666667  
..         ...   ...       ...    ...        ...  
647       63.4  55.2      81.0   71.8  61.833333  
648       65.0  64.2      83.2   65.0  67.333333  
649       66.0  67.0      80.0   71.4  64.166667  
650       60.4  72.2      83.4   71.6  63.666667  
651       51.0  44.6      60.2   62.6  52.166667  

[652 rows x 28 columns]

<p>Посмотрим на рандомного игрока:</p>

In [36]:
data.iloc[np.random.randint(data.shape[0])]

short_name                                                  N. González
long_name                                         Nicolás Iván González
overall                                                              77
age                                                                  23
dob                                                          1998-04-06
height_cm                                                           180
weight_kg                                                            72
club_position                                                        LW
Position                                                        Forward
club_name                                                    Fiorentina
league_name                                             Italian Serie A
nationality_name                                              Argentina
club_jersey_number                                                   22
preferred_foot                                                  

<p>Этот набор данных включает данные о 650 лучших игроках В игре FIFA 22.<br>
Набор данных привлекает возможность сравнивать показатели видеоигры с реальной жизнью.<br>
Сравнивая виртуальную жизнь с реальной, можно понять степерь реальности видеоигры.<br>
С помощью этого набора данных можно выявить лучшие навыки для игрока в футбол и собрать <br>
самую подходящую команду.</p>

<ul>Также с помощью этого набора можно ответить на вопросы:
<li>Какая преобладающая позиция среди лучший игроков?
<li>Какие страны наиболее представлены в данных?
<li>Каковы наиболее распространенные позиции игроков?
<li>Кто лучшие игроки в целом?
</ul>


<h1>Анализ данных</h1>

<p>Проверим данные на пропуски</p>

In [66]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 652 entries, 0 to 651
Data columns (total 28 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   short_name          652 non-null    object 
 1   long_name           652 non-null    object 
 2   overall             652 non-null    int64  
 3   age                 652 non-null    int64  
 4   dob                 652 non-null    object 
 5   height_cm           652 non-null    int64  
 6   weight_kg           652 non-null    int64  
 7   club_position       652 non-null    object 
 8   Position            652 non-null    object 
 9   club_name           652 non-null    object 
 10  league_name         652 non-null    object 
 11  nationality_name    652 non-null    object 
 12  club_jersey_number  652 non-null    int64  
 13  preferred_foot      652 non-null    object 
 14  weak_foot           652 non-null    int64  
 15  skill_moves         652 non-null    int64  
 16  player_t

In [67]:
data.isnull().sum()

short_name            0
long_name             0
overall               0
age                   0
dob                   0
height_cm             0
weight_kg             0
club_position         0
Position              0
club_name             0
league_name           0
nationality_name      0
club_jersey_number    0
preferred_foot        0
weak_foot             0
skill_moves           0
player_traits         0
pace                  0
shooting              0
passing               0
dribbling             0
defending             0
physic                0
attacking             0
skill                 0
movement              0
power                 0
mentality             0
dtype: int64

<p>Найдем числовые и категориальные данные</p>

In [47]:
numerical_columns = [col for col in data.columns
                     if data[col].dtypes != 'object']

categorical_columns = [col for col in data.columns
                       if data[col].dtypes == 'object']

data.dtypes

short_name             object
long_name              object
overall                 int64
age                     int64
dob                    object
height_cm               int64
weight_kg               int64
club_position          object
Position               object
club_name              object
league_name            object
nationality_name       object
club_jersey_number      int64
preferred_foot         object
weak_foot               int64
skill_moves             int64
player_traits          object
pace                  float64
shooting              float64
passing               float64
dribbling             float64
defending             float64
physic                float64
attacking             float64
skill                 float64
movement              float64
power                 float64
mentality             float64
dtype: object

<p>Рассчитаем максимум среди каждого числового признака</p>

In [60]:
data[numerical_columns].max()

#Вывести игроков, которые имеют максимальное значение признака
#for i in numerical_columns:
    #print(data[data[i] == data[i].max()])

overall                93.0
age                    43.0
height_cm             201.0
weight_kg             100.0
club_jersey_number     99.0
weak_foot               5.0
skill_moves             5.0
pace                   97.0
shooting               94.0
passing                93.0
dribbling              95.0
defending              91.0
physic                 90.0
attacking              87.6
skill                  94.0
movement               92.4
power                  87.2
mentality              83.5
dtype: float64

<p>Рассчитаем минимум среди каждого числового признака</p>

In [61]:
data[numerical_columns].min()

#Вывести игроков, которые имеют минимальное значение признака
#for i in numerical_columns[:2]:
    #print(data[data[i] == data[i].max()])

overall                77.000000
age                    18.000000
height_cm             158.000000
weight_kg              55.000000
club_jersey_number      1.000000
weak_foot               1.000000
skill_moves             1.000000
pace                   32.000000
shooting               25.000000
passing                44.000000
dribbling              49.000000
defending              20.000000
physic                 45.000000
attacking               8.600000
skill                  14.400000
movement               31.000000
power                  37.200000
mentality              21.833333
dtype: float64

Рассчитаем среднее значение каждого числового признака

In [50]:
data[numerical_columns].mean()

overall                80.757669
age                    27.776074
height_cm             181.940184
weight_kg              76.455521
club_jersey_number     14.605828
weak_foot               3.274540
skill_moves             2.981595
pace                   74.065505
shooting               68.272597
passing                73.292443
dribbling              77.007712
defending              62.009510
physic                 73.087370
attacking              62.432822
skill                  65.394479
movement               72.036503
power                  69.651534
mentality              65.086656
dtype: float64

Рассчитаем стандартное отклонение каждого числового признака

In [70]:
data[numerical_columns].std()

overall                3.214242
age                    4.133581
height_cm              7.100623
weight_kg              7.531777
club_jersey_number    13.841416
weak_foot              0.775842
skill_moves            1.051500
pace                  10.352396
shooting              12.543588
passing                7.350904
dribbling              7.063509
defending             17.394831
physic                 7.723627
attacking             18.308762
skill                 18.803734
movement              10.891396
power                  9.959120
mentality             13.579516
dtype: float64

<p>Рассчитаем медиану каждого числового признака</p>

In [71]:
data[numerical_columns].median()

overall                80.000000
age                    28.000000
height_cm             182.000000
weight_kg              76.000000
club_jersey_number     10.000000
weak_foot               3.000000
skill_moves             3.000000
pace                   73.898894
shooting               70.000000
passing                73.000000
dribbling              78.000000
defending              63.000000
physic                 73.292259
attacking              68.600000
skill                  71.700000
movement               74.400000
power                  72.200000
mentality              69.000000
dtype: float64

<p>Общая информация</p>

In [68]:
data.describe()

overall         age   height_cm   weight_kg  club_jersey_number  \
count  652.000000  652.000000  652.000000  652.000000          652.000000   
mean    80.757669   27.776074  181.940184   76.455521           14.605828   
std      3.214242    4.133581    7.100623    7.531777           13.841416   
min     77.000000   18.000000  158.000000   55.000000            1.000000   
25%     78.000000   25.000000  177.000000   71.000000            6.000000   
50%     80.000000   28.000000  182.000000   76.000000           10.000000   
75%     82.000000   30.000000  187.000000   82.000000           21.000000   
max     93.000000   43.000000  201.000000  100.000000           99.000000   

        weak_foot  skill_moves        pace    shooting     passing  \
count  652.000000   652.000000  652.000000  652.000000  652.000000   
mean     3.274540     2.981595   74.065505   68.272597   73.292443   
std      0.775842     1.051500   10.352396   12.543588    7.350904   
min      1.000000     1.000000   32.000000   25.000000   44.000000   
25%      3.000000     2.000000   69.000000   64.000000   70.000000   
50%      3.000000     3.000000   73.898894   70.000000   73.000000   
75%      4.000000     4.000000   81.000000   77.000000   78.000000   
max      5.000000     5.000000   97.000000   94.000000   93.000000   

        dribbling   defending      physic   attacking       skill    movement  \
count  652.000000  652.000000  652.000000  652.000000  652.000000  652.000000   
mean    77.007712   62.009510   73.087370   62.432822   65.394479   72.036503   
std      7.063509   17.394831    7.723627   18.308762   18.803734   10.891396   
min     49.000000   20.000000   45.000000    8.600000   14.400000   31.000000   
25%     75.000000   45.000000   69.000000   60.150000   62.550000   65.600000   
50%     78.000000   63.000000   73.292259   68.600000   71.700000   74.400000   
75%     82.000000   77.000000   79.000000   73.400000   77.200000   80.200000   
max     95.000000   91.000000   90.000000   87.600000   94.000000   92.400000   

            power   mentality  
count  652.000000  652.000000  
mean    69.651534   65.086656  
std      9.959120   13.579516  
min     37.200000   21.833333  
25%     65.750000   63.500000  
50%     72.200000   69.000000  
75%     76.600000   73.166667  
max     87.200000   83.500000